src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java

```diff
@@ -949,4 +953,28 @@ private void processSearchComposite(final SearchCompositeDto<ObservationUnitsSea
 			searchDTO.setSearchRequest(searchRequest);
 		}
 	}
+
+	private void correctKSUDateFormatIfNecessary(final Table<String, String, String> table, final List<MeasurementVariable> measurementVariables) {
+
+		final List<MeasurementVariable> dateVariables = measurementVariables.stream().filter(measurementVariable -> measurementVariable.getDataTypeId()!=null && measurementVariable.getDataTypeId() == TermId.DATE_VARIABLE
+			.getId()).collect(Collectors.toList());
+		if (dateVariables != null) {
+			final List<String> dateMeasurementVariables = dateVariables.stream().map(measurementVariable -> measurementVariable.getName()).collect(Collectors.toList());
+			if (dateMeasurementVariables != null) {
+				for (final String obsUnit : table.rowKeySet()) {
+					for (final String colVariable : table.columnKeySet()) {
```

Comment:
Switch the order of the two for loops so that you don't unnecessariy go through all the observations if the column is not a date variable